pkg/columns/formatter/json/json.go

```diff
@@ -0,0 +1,334 @@
+					return
+				}
+				e.WriteString("false")
+			}
+		case reflect.Float32:
+			ff := columns.GetFieldAsNumberFunc[float64, T](col)
+			formatter = func(e *encodeState, t *T) {
+				e.Write(key)
+				floatEncoder(32).writeFloat(e, ff(t))
+			}
+		case reflect.Float64:
+			ff := columns.GetFieldAsNumberFunc[float64, T](col)
+			formatter = func(e *encodeState, t *T) {
+				e.Write(key)
+				floatEncoder(64).writeFloat(e, ff(t))
+			}
+		case reflect.Array:
+			ff := columns.GetFieldAsString[T](col)
+			formatter = func(e *encodeState, t *T) {
+				e.Write(key)
+				writeString(e, ff(t))
+			}
+		case reflect.String:
+			ff := columns.GetFieldFunc[string, T](col)
+			formatter = func(e *encodeState, t *T) {
+				e.Write(key)
+				writeString(e, ff(t))
+			}
+		}
+
+		ncols = append(ncols, &column[T]{
+			column:    col,
+			formatter: formatter,
+		})
+	}
+
+	tf := &Formatter[T]{
+		options: opts,
+		columns: ncols,
+	}
+	return tf
+}
+
+func (f *Formatter[T]) printCols(entry *T, buf *encodeState, cols []*column[T], level int) {
+	indent := strings.Repeat("  ", level+1)
+	colsWithParent := map[string][]*column[T]{}
+	colsWithoutParent := []*column[T]{}
+
+	for _, col := range cols {
+		// Save columns which have a parent in its name
+		if strings.Count(col.column.Name, ".") > level {
+			hierarchy := strings.Split(col.column.Name, ".")
+			parentName := hierarchy[level]
+			colsWithParent[parentName] = append(colsWithParent[parentName], col)
+			continue
+		}
+		// Don't print columns without a parent directly
+		// It is possible that their name is the same as a name of an object in the same level
+		// For example: {"a": {"bbb": 1}, "a": 2}
+		// We will filter them out after we gathered all object names
+		colsWithoutParent = append(colsWithoutParent, col)
+	}
+
+	first := true
+	// Better name?
+	handlePrefix := func() {
+		if !first {
+			buf.WriteByte(',')
+			if f.options.prettyPrint {
+				buf.WriteString("\n" + indent)
+			} else {
+				buf.WriteByte(' ')
+			}
+		} else {
+			if f.options.prettyPrint {
+				buf.WriteString("\n" + indent)
+			}
+			first = false
+		}
+	}
+
+	// Now print columns with their parents
+	for key, val := range colsWithParent {
+		handlePrefix()
+
+		parentName, _ := json.Marshal(strings.Split(key, ".")[0])
+		buf.Write(parentName)
+		buf.WriteString(": {")
+
+		f.printCols(entry, buf, val, level+1)
+
+		// End parent
+		if f.options.prettyPrint {
+			buf.WriteString("\n" + indent + "}")
+		} else {
+			buf.WriteByte('}')
+		}
+	}
+
+	// Now all cols without a parent
+	for _, col := range colsWithoutParent {
+		childName := strings.Split(col.column.Name, ".")[level]
+		_, found := colsWithParent[childName]
+		if found {
+			// We already have an object with this key, skip this column
+			continue
+		}
+		handlePrefix()
+		col.formatter(buf, entry)
+	}
+}
+
+// FormatEntry returns an entry as a formatted string, respecting the given formatting settings
+func (f *Formatter[T]) FormatEntry(entry *T) string {
```

Comment:
While experimenting with snapshotter support in containerized gadgets, I found out that a `FormatEntries` function will be useful. Let's keep going with this PR as it's but keep in mind that'd be needed very soon.